L1Trigger/L1THGCal/interface/hw_interfaces/HGCalHardwareInterface.h

```diff
@@ -0,0 +1,59 @@
+#ifndef __L1Trigger_L1THGCal_HGCalHardwareInterface_h__
+#define __L1Trigger_L1THGCal_HGCalHardwareInterface_h__
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/L1THGCal/interface/HGCalCluster.h"
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+class HGCalHardwareInterface{
+  
+ public:
+  /* mapping is retrieved from: https://twiki.cern.ch/twiki/bin/viewauth/CMS/HGCALTriggerArchitectureDesignAndImplementation */
+  struct ClusterDataFormat{
+    enum CDF {ET = 0, CenterY = 8, CenterX = 20, Flags = 32, SizeY = 38, SizeX = 46, LocalMax = 54, NoTriggerCells = 56, LocalMax1RelX = 64,  LocalMax0ET = 72, LocalMax0RelY = 80, LocalMax0RelX = 88, LocalMax2RelY = 96,  LocalMax2RelX = 104,  LocalMax1ET = 112,  LocalMax1RelY = 120,  LocalMax3ET = 128, LocalMax3RelY = 136,  LocalMax3RelX = 144,  LocalMax2ET = 152 };
+  };
+  struct MulticlusterDataFormat{
```

Comment:
I agree that right now we lack information on the hardware-layer origin. This is due to the fact that all the backend algos are grouped in the same place and the collections are produced by the same producer, and it's not really easy to attach collections to hardware layers.
But in the future proposed code architecture (you can have a look [here](https://indico.cern.ch/event/624663/contributions/2525557/attachments/1431967/2200062/17.03.21_emulator.postTDR.pdf)) there will be a clear separation of the processing layers (with different producers). And each collection will be attached to a specific hardware layer.

Is it in line with what you are thinking about, or were you thinking maybe at encoding directly in the objects themselves the layer where they come from?